lib/decoders/shapefile.js

```diff
@@ -57,12 +58,14 @@ class Shapefile extends Duplex {
     return `/tmp/${name}_${this._fgroup}${extension}`;
   }
 
-  write(chunk, encoding, done) {
-    logger.debug(`Writing shape feature with ${encoding} encoding`);
-    return this._zBuffer.write(chunk, encoding, done);
+  _write(chunk, encoding, done) {
+    logger.debug(`Writing shape feature chunk ${typeof chunk} with ${encoding} encoding`);
+    return this._zBuffer.write(new Buffer(chunk.toString('binary'), 'binary'), 'binary', done);
```

Comment:
an explanatory comment of why this strange thing is necessary would be nice. required by gzip file format or what?